libraries/chain/htlc_evaluator.cpp

```diff
@@ -48,6 +48,13 @@ namespace graphene {
          FC_ASSERT( o.preimage_size <= htlc_options->max_preimage_size, "HTLC preimage length exceeds allowed length" );
          // make sure the sender has the funds for the HTLC
          FC_ASSERT( d.get_balance( o.from, o.amount.asset_id) >= (o.amount), "Insufficient funds") ;
+         // memo field added at harfork BSIP64
+         // NOTE: this check can be removed after hardfork time
+         FC_ASSERT( d.head_block_time() > HARDFORK_CORE_BSIP64_TIME || !o.extensions.value.memo.valid(), 
+               "Memo unavailable until after HARDFORK BSIP64");
```

Comment:
I guess you didn't add code for proposals?

Reply:
I am sorry, I am unclear as to what you are asking. Are you saying additional hardfork logic is required for proposals? Perhaps supply a scenario, and my lightbulb will hopefully come on.

Reply:
Something similar to this: https://github.com/bitshares/bitshares-core/blob/53f0a00c4c7f7832ced1b9e690a405def239f007/libraries/chain/proposal_evaluator.cpp#L48-L55.